python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py

```diff
@@ -264,42 +352,92 @@ def _create_streaming_chat_message_content(
-                continue
+    def _verify_function_choice_behavior(self, settings: AzureAIInferenceChatPromptExecutionSettings, **kwargs: Any):
+        """Verify the function choice behavior."""
+        kernel = kwargs.get("kernel", None)
+        arguments = kwargs.get("arguments", None)
+
+        if settings.function_choice_behavior is not None:
+            if kernel is None:
+                raise ServiceInvalidExecutionSettingsError("Kernel is required for tool calls.")
+            if arguments is None and settings.function_choice_behavior.auto_invoke_kernel_functions:
+                raise ServiceInvalidExecutionSettingsError("Kernel arguments are required for auto tool calls.")
+            if settings.extra_parameters is not None and settings.extra_parameters.get("n", 1) > 1:
+                # Currently only OpenAI models allow multiple completions but the Azure AI Inference service
+                # does not expose the functionality directly. If users want to have more than 1 responses, they
+                # need to configure `extra_parameters` with a key of "n" and a value greater than 1.
+                raise ServiceInvalidExecutionSettingsError(
+                    "Auto invocation of tool calls may only be used with a single completion."
+                )
 
-            # If it's a user message and there are any image items in the message, we need to create a list of
-            # content items, otherwise we need to just pass in the content as a string or it will error.
-            contentItems = []
-            for item in message.items:
-                if isinstance(item, TextContent):
-                    contentItems.append(TextContentItem(text=item.text))
-                elif isinstance(item, ImageContent) and (item.data_uri or item.uri):
-                    contentItems.append(
-                        ImageContentItem(
-                            image_url=ImageUrl(url=item.data_uri or str(item.uri), detail=ImageDetailLevel.Auto)
-                        )
-                    )
-                else:
-                    logger.warning(
-                        "Unsupported item type in User message while formatting chat history for Azure AI"
-                        f" Inference: {type(item)}"
-                    )
-            chat_request_messages.append(_MESSAGE_CONVERTER[message.role](content=contentItems))
+    def _configure_function_choice_behavior(
+        self, settings: AzureAIInferenceChatPromptExecutionSettings, kernel: Kernel
+    ):
+        """Configure the function choice behavior to include the kernel functions."""
+
+        def _config_call_back(
+            function_choice_configuration: FunctionCallChoiceConfiguration,
+            settings: AzureAIInferenceChatPromptExecutionSettings,
+            type: str,
+        ):
+            """Update the settings from a FunctionChoiceConfiguration."""
+            if function_choice_configuration.available_functions:
```

Comment:
This code already exists in a `function_calling_utils` file. Why is there a need to duplicate it here?

Reply:
agreed, since the format and names are the same, this should use a generic function

Reply:
The settings parameters are of different types for different services.

Reply:
FunctionChoiceBehavior is on the PromptExecutionSettings, and doesn't differ in types between different services. There shouldn't be a need to have this duplicated between services.